xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-ui/src/main/resources/AnnotationCode/Script.xml

```diff
@@ -331,10 +331,15 @@ XWiki.Selection = Class.create({
     return descendant;
   },
 
+  // Removes the highligh on selected text (e.g., after pressing when annotating a new piece of text).
   removeSelectionHighlight : function() {
-    // unwrap
+    // We used previously replace from prototype earlier, but it behaves badly when the parent of the wrapper is a td
+    // (moving the selected text at the end of the td).
+    // Also, wrapper is always an element containing only text. If &lt;strong&gt;a&lt;em&gt;b&lt;/em&gt;c&lt;/strong&gt; is
+    // selected for annotation, a, b, and c will each be wrapped in a &lt;span class='selection-highligh' ...&gt;.
+    // Therefore, when unwrapping we know that the spans of this.highlightWrappers contain only text.
     this.highlightWrappers.each(function(wrapper) {
-      wrapper.replace(wrapper.innerHTML);
+      wrapper.replaceWith(wrapper.innerHTML);
```

Comment:
This seems wrong. You're assigning `innerHTML` that, e.g., escapes `<` as `&lt;` to the text content which would then display this escaped text literally. Why don't you use something like `wrapper.replaceWith(...wrapper.childNodes)` (untested)?

Reply:
You are right, highlighting and cancelling `a&amp;b` would replace it with the raw content instead of html.
Thanks for catching this (and your snippet is correct).

Reply:
I have a patch for this on an XWiki instance I have, I just need to isolate it and propose it here.